com.unity.shadergraph/Editor/Data/Graphs/SerializableCubemap.cs

```diff
@@ -23,6 +23,42 @@ class CubemapHelper
 #pragma warning restore 649
         }
 
+        // used to get a Cubemap ref guid without loading the cubemap asset itself into memory
+        [Serializable]
+        class MinimalCubemapHelper
+        {
+            [Serializable]
+            public struct MinimalTextureRef
+            {
+                public string guid;
+            }
+            public MinimalTextureRef cubemap;
+        }
+
+        internal string guid
```

Comment:
Can we genericize this function so we don't have effectively 3 copies?

Reply:
Unfortunately no (not easily), because the names of the variables are different :(

Reply:
Ideally we would switch to using a generic implementation, but I'll leave that for the new graph work, don't want to get into redefining the data structures in a bug fix.  :)

Reply:
Really should be a "SerializableAssetRef<T>" generic class.   Would likely require the addition of more versioning code to make that switch though..  we should just make sure the new graph nodes use something like that, and we can apply the versioning as part of the upgrade.